airbyte-ci/connectors/pipelines/pipelines/dagger/actions/gradle.py

```diff
@@ -0,0 +1,230 @@
+                )
+            )
+            # Set RUN_IN_AIRBYTE_CI to tell gradle how to configure its build cache.
+            # This is consumed by settings.gradle in the repo root.
+            .with_env_variable("RUN_IN_AIRBYTE_CI", "1")
+            # Disable the Ryuk container because it needs privileged docker access which it can't have.
+            .with_env_variable("TESTCONTAINERS_RYUK_DISABLED", "true").with_workdir(self.workdir)
+            # Ensure that the .m2 directory exists.
+            .with_exec(["mkdir", "-p", self.LOCAL_MAVEN_REPOSITORY_PATH])
+        )
+
+    @staticmethod
+    def set_s3_build_cache_credentials(
+        s3_build_cache_access_key_id: Optional[dagger.Secret], s3_build_cache_secret_key: Optional[dagger.Secret]
+    ) -> Callable:
+        """Closure to set the S3 build cache credentials as environment variables on a gradle container.
+
+        Args:
+            s3_build_cache_access_key_id (Optional[dagger.Secret]): The S3 build cache access key ID as a dagger secret.
+            s3_build_cache_secret_key (Optional[dagger.Secret]): The S3 build cache secret key as a dagger secret.
+
+        Returns:
+            Callable: A function which takes a gradle container and returns a gradle container with the S3 build cache credentials set as environment variables.
+        """
+
+        def wrapper(gradle_container: dagger.Container) -> dagger.Container:
+            if s3_build_cache_access_key_id:
+                gradle_container = gradle_container.with_secret_variable("S3_BUILD_CACHE_ACCESS_KEY_ID", s3_build_cache_access_key_id)
+            if s3_build_cache_secret_key:
+                gradle_container = gradle_container.with_secret_variable("S3_BUILD_CACHE_SECRET_KEY", s3_build_cache_secret_key)
+            return gradle_container
+
+        return wrapper
+
+    def warm_up_dependency_cache(
+        self, gradle_dep_cache_path: str, gradle_home_path: str, build_cdk: bool, warm_dependency_cache_args: Tuple[str] = ("--dry-run")
+    ) -> Callable:
+        """Closure to seed the gradle home cache from the cache volume.
+        Args:
+            gradle_dep_cache_path (str): The path to the gradle dependency cache volume mount.
+            gradle_home_path (str): The path to the gradle home directory.
+            build_cdk (bool): Whether to build the CDK.
+            local_maven_repository_path (str): The path to the local maven repository.
+
+        Returns:
+            Callable: A function which takes a gradle container and returns a gradle container with the gradle cache seeded from the volume.
+        """
+
+        rsync_cache_volume_to_gradle_home = f"(rsync -a --stats --mkpath {gradle_dep_cache_path}/ {gradle_home_path} || true)"
+        rsync_gradle_home_to_cache_volume = f"(rsync -a --stats {self.GRADLE_HOME_PATH}/ {self.GRADLE_DEP_CACHE_PATH} || true)"
+        # Running a gradle task like "help" with these arguments will trigger updating all dependencies.
```

Comment:
Actually, it's `--write-verification-metadata --dry-run` which warms the cache. Gradle's CLI kind of sucks here. 

Best move the `warm_dependency_cache_args = ["--write-verification-metadata", "sha256", "--dry-run"] if not self.local_execution else ["--dry-run"]` in here, the last thing we want is people customizing this stuff. The whole point of this change is to re-use the same container for all gradle tasks, after all, and only the execution context should change things.